### PR TITLE
feat(agent): Phase 2 - LLM Intelligence for Agent Control Loop

### PR DESCRIPTION
**Pull request by DominicBurkart**
## Summary

Implements Phase 2 of the agent architecture by adding LLM-driven decision making to the agent control loop. This replaces deterministic MVP heuristics with intelligent reasoning while maintaining full backward compatibility.

## Stack Position

**PR Stack:**
1. PR #37: Phase 1 - MVP Control Loop → `main` 
2. **THIS PR: Phase 2 - LLM Intelligence** → PR #37 (`feat/agent-mvp-control-loop`)
3. PR #39: Phase 3 - vLLM Infrastructure → THIS PR

### 1. **New LLM Prompt Module** (`harness/src/agent/prompts.rs`)
- **PlanningPrompt**: Asks LLM to analyze user request and create execution plan
- **DecisionPrompt**: Asks LLM to decide "QUERY" (need RAG) vs "PROCEED" (ready to act)
- **CompletionPrompt**: Asks LLM to determine "COMPLETE" vs "INCOMPLETE"
- Includes comprehensive parsing and validation
- 25 unit tests covering all prompt types and edge cases

### 2. **Enhanced Agent Architecture** (`harness/src/agent/mod.rs`)
- Added `llm_provider: Option<Arc<dyn ModelProvider>>` field
- Added `plan_cache: Option<String>` to store LLM-generated plans
- New `with_llm()` constructor for LLM-enabled agents
- Backward compatible: existing constructors create MVP-mode agents

### 3. **LLM-Powered Methods**
- **`plan()`**: Uses LLM to analyze context and create execution plan (when available)
- **`decide()`**: Uses LLM to intelligently decide whether to query RAG or proceed
- **`check_task_complete()`**: Uses LLM to validate task completion based on user intent
- All methods fall back to MVP mode when LLM unavailable

### 4. **Robust Error Handling**
- **`call_llm_with_retry()`**: Exponential backoff retry logic (max 3 attempts)
- **`validate_llm_response()`**: Validates LLM output before using it
- Graceful fallbacks when LLM fails or produces invalid responses
- Comprehensive logging for debugging

### 5. **Comprehensive Testing**
- ✅ 73 unit tests pass (including 25 new prompt tests)
- ✅ Full LLM integration test (`test_full_llm_agent_control_loop`)
- ✅ Backward compatibility test (`test_mvp_mode_still_works_without_llm`)
- ✅ All existing MVP tests continue to pass
- ✅ LLM tests skip gracefully when Ollama unavailable

## Architecture Decisions

1. **Model Selection**: Uses `qwen2.5:0.5b` for fast, lightweight inference
2. **Temperature Settings**: 
   - Planning: 0.7 (creative)
   - Decision: 0.3 (conservative)
   - Completion: 0.2 (deterministic)
3. **Fallback Strategy**: Gracefully degrades to MVP mode when LLM unavailable
4. **Validation**: Keywords-based validation (QUERY/PROCEED, COMPLETE/INCOMPLETE)

## Backward Compatibility

✅ **Fully backward compatible**:
- Existing MVP tests pass without modification
- Agents created with `new()` or `with_entity_store()` work in MVP mode
- LLM functionality only activated when using `with_llm()` constructor

## Example Usage

```rust
use model::OllamaProvider;
use std::sync::Arc;

// Create LLM-enabled agent
let provider = Arc::new(OllamaProvider::with_default_config()?);
let mut agent = AgentLoop::with_llm(config, entity_store, provider);

// Run agent - uses LLM for intelligent decision making
let result = agent.run(context).await?;
```

## Commits

- `84c14f1` feat: add LLM intelligence to agent control loop (Phase 2)
- `ba5d2df` fix: resolve critical bugs in LLM integration
- `1f4fa0f` fix(deny): resolve cargo-deny license warnings and CVSS 4.0 compatibility

## Next Steps

After this PR merges to PR #37:
- Phase 3: vLLM container infrastructure (PR will follow)
- Phase 4: Update harness container integration
- Phase 5: Integration tests
- Phase 6: CI workflows

### CODE REVIEW

File: harness/src/agent/mod.rs [hunk 1]
```diff
@@ -227,27 +324,196 @@ impl AgentLoop {
             }
         }
 
+        // If LLM available, use it for planning
```
Comment (DominicBurkart) [inline harness/src/agent/mod.rs]: After this PR is merged, LLM availability should be a hard requirement. How can the system be simplified if we require (and fail if we cannot initialize) the associated LLM container? 

Critically, how can we encode a fail-fast philosophy with descriptive errors for each individual failure condition into the program, such that integration tests and/or agent evals can effectively capture both happy paths and known failure conditions with specific, concise, agent-actionable errors?
